docling/models/vlm_models_inline/hf_transformers_model.py

```diff
@@ -128,7 +128,11 @@ def __call__(
                     )
 
                     # Define prompt structure
-                    prompt = self.formulate_prompt()
+                    if callable(self.vlm_options.prompt):
+                        user_prompt = self.vlm_options.prompt(page)
```

Comment:
Pass `page.parsed_page` here when using `SegmentedPage` in the callable signature.